### PR TITLE
Feature/generic staging

### PR DESCRIPTION
**Pull request by marksantcroos**
These changes generalize the file staging which allows it to work with non-ssh transfers, e.g. Globus Online, and it also leads the way for OSG development.

### CODE REVIEW

File: src/radical/pilot/configs/nersc.json [hunk 1]
```diff
@@ -76,7 +80,7 @@
         "default_remote_workdir"      : "$SCRATCH",
         "valid_roots"                 : ["/global", "/scratch1", "/scratch2"],
         "bootstrapper"                : "default_bootstrapper.sh",
-        "rp_version"                  : "debug",
+        "rp_version"                  : "local",
```
Comment (andre-merzky) [inline src/radical/pilot/configs/nersc.json]: Is this intentional?

File: src/radical/pilot/controller/input_file_transfer_worker.py [hunk 2]
```diff
@@ -174,22 +171,16 @@ def run(self):
...
                                 else:
                                     copy_flags = 0
 
-                                try :
-                                    input_file.copy(target, flags=copy_flags)
-                                except Exception as e :
-                                    logger.exception (e)
-                                input_file.close()
+                                try:
+                                    saga_dir.copy(input_file_url, target, flags=copy_flags)
```
Comment (andre-merzky) [inline src/radical/pilot/controller/input_file_transfer_worker.py]: input staging now is done on the saga_dir object -- shouldn't the same be applied to the output staging, or is that not needed there / already there differently?

**Comment by andre-merzky** [comment]
I did not test the branch, but otherwise this looks ok to me (modulo the comment on output-FTW.

File: src/radical/pilot/configs/nersc.json (hunk 1 above)
Comment (marksantcroos) [inline src/radical/pilot/configs/nersc.json]: As discussed, sort of, as its the default anyway.

**Comment by andre-merzky** [comment]
Many thanks, Mark! :D

**Comment by marksantcroos** [comment]
Do I get a cookie now? :)

**Pull request merged**